### PR TITLE
SPLAT-2864: Promote VSphereMultiVCenterDay2 to GA

### PR DESCRIPTION
**Pull request by vr4manta**
[SPLAT-2864](https://redhat.atlassian.net/browse/SPLAT-2864)

### Changes
- Promoted VSphereMultiVCenterDay2 to GA

### CODE REVIEW

File: config/v1/tests/infrastructures.config.openshift.io/AAA_ungated.yaml [hunk 1]
```diff
@@ -1884,6 +1883,8 @@ tests:
       initialCRDPatches:
         - op: remove
           path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/platformSpec/properties/vsphere/properties/vcenters/minItems
+        - op: remove
+          path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/platformSpec/properties/vsphere/x-kubernetes-validations/0
```
Comment (saschagrunert) [inline config/v1/tests/infrastructures.config.openshift.io/AAA_ungated.yaml:1887]: nit: patching by array index (`x-kubernetes-validations/0`) is fragile; if the rule order changes in a future generation, this silently patches out the wrong rule. Not a blocker since it matches the current generated output, but worth noting.

**Review by saschagrunert — APPROVED** [review]
CEL validation rules are correct across all edge cases (pure adds, removes, swaps, simultaneous add+remove, no-change updates, failure domain cross-validation). Promotion mechanics are sound, ratcheting is handled properly.

One minor nit: `config/v1/types_infrastructure.go:1699` still says "in TechPreview" in the vcenters godoc, but the field is now Default.

File: config/v1/tests/infrastructures.config.openshift.io/AAA_ungated.yaml (hunk 1 above)
Comment (vr4manta) [reply config/v1/tests/infrastructures.config.openshift.io/AAA_ungated.yaml:1887]: i agree.  i wasn't sure of a way to patch it out for ratcheting tests.  Is there something better i can do for this test?

**Comment by vr4manta** [comment]
> CEL validation rules are correct across all edge cases (pure adds, removes, swaps, simultaneous add+remove, no-change updates, failure domain cross-validation). Promotion mechanics are sound, ratcheting is handled properly.
> 
> One minor nit: `config/v1/types_infrastructure.go:1699` still says "in TechPreview" in the vcenters godoc, but the field is now Default.

I'll update the infrastructure go file to no longer mention TechPreview.

File: config/v1/types_infrastructure.go [hunk 2]
```diff
@@ -1703,12 +1703,9 @@ type VSpherePlatformNodeNetworking struct {
 type VSpherePlatformSpec struct {
 	// vcenters holds the connection details for services to communicate with vCenter.
 	// Up to 3 vCenters are supported.
-	// Once the cluster has been installed, you are unable to change the current number of defined
-	// vCenters except when 1.) the cluster has been upgraded from a version of OpenShift
-	// where the vsphere platform spec was not present or 2.) in TechPreview you are able to add and
-	// remove vCenters but may not remove all vCenters.  You may make modifications to the existing
-	// vCenters that are defined in the vcenters list in order to match with any added or modified
-	// failure domains.
+	// After installation, you can add or change vCenters, or remove some of them, but you must keep
```
Comment (saschagrunert) [inline config/v1/types_infrastructure.go:1727]: The godoc describes that users can "add or change vCenters, or remove some of them" but does not mention that these are mutually exclusive per-update operations. The FeatureGateAwareXValidation rules on lines 1709-1710 enforce that adds and removes cannot happen in the same update, e.g. `[A,B] -> [A,C]` (swap B for C) is rejected with "Cannot add and remove vCenters at the same time".

Consider adding something like: "Adding and removing vCenters must be done as separate updates."

Comment (vr4manta) [reply config/v1/types_infrastructure.go:1727]: i'll make that change.

**Comment by vr4manta** [comment]
Need to rebase and get this new featuregate variant updated.

File: config/v1/types_infrastructure.go [hunk 3]
```diff
@@ -1722,12 +1722,10 @@ type VSpherePlatformNodeNetworking struct {
...
+	// at least one and may not add and remove vCenters during the same update.  You may make modifications
+	// to the existing vCenters that are defined in the vcenters list in order to match with any added or
+	//modified failure domains.
```
Comment (saschagrunert) [inline config/v1/types_infrastructure.go]: Missing space after `//`: `//modified` should be `// modified`. This propagates to all generated CRD descriptions and swagger docs.

File: features/features.go [hunk 4]
```diff
@@ -814,7 +814,7 @@ var (
 						contactPerson("vr4manta").
 						productScope(ocpSpecific).
 						enhancementPR("https://github.com/openshift/enhancements/pull/1961").
-						enable(inTechPreviewNoUpgrade(), inDevPreviewNoUpgrade()).
+						enable(inDefault(), inOKD(), inTechPreviewNoUpgrade(), inDevPreviewNoUpgrade()).
```
Comment (saschagrunert) [inline features/features.go:793]: `verify-feature-promotion` is failing: `[sig-cluster-lifecycle][OCPFeatureGate:VSphereMultiVCenterDay2][platform:vsphere] vSphere failure domain ValidatingAdmissionPolicies should block removing a failure domain referenced by a Machine` has a 72% pass rate (needs >= 95%). This blocks promotion.

Comment (vr4manta) [reply features/features.go:793]: I can run some more.  Its odd cause i see 84 percent in sippy.

**Comment by vr4manta** [comment]
/assign @saschagrunert 
Changes are made and promotion presubmit passing.  Ready for final review.

**Comment by vr4manta** [comment]
Now feature promotion job is failing since we did branching.  its looking at tests in 5.1 so the history of all the jobs i ran to get numbers up are now useless data.

File: config/v1/types_infrastructure.go [hunk 5]
```diff
@@ -1725,12 +1725,10 @@ type VSpherePlatformNodeNetworking struct {
...
 	// vcenters holds the connection details for services to communicate with vCenter.
 	// Up to 3 vCenters are supported.
-	// Once the cluster has been installed, you are unable to change the current number of defined
-	// vCenters except when 1.) the cluster has been upgraded from a version of OpenShift
-	// where the vsphere platform spec was not present or 2.) in TechPreview you are able to add and
-	// remove vCenters but may not remove all vCenters.  You may make modifications to the existing
-	// vCenters that are defined in the vcenters list in order to match with any added or modified
-	// failure domains.
+	// After installation, you can add or change vCenters, or remove some of them, but you must keep
+	// at least one and may not add and remove vCenters during the same update.  You may make modifications
```
Comment (saschagrunert) [inline config/v1/types_infrastructure.go:1728]: nit:
```suggestion
	// at least one and may not add and remove vCenters during the same update. You may make modifications
```

**Review by saschagrunert — APPROVED** [review]
LGTM

**Review by everettraven — COMMENTED** [review]
Changes here LGTM, but verify-feature-promotion has a few tests that are _just_ below the required number of runs threshold.

Is it possible for us to snag a couple more runs?

**Comment by vr4manta** [comment]
> Changes here LGTM, but verify-feature-promotion has a few tests that are _just_ below the required number of runs threshold.
> 
> Is it possible for us to snag a couple more runs?

I started 3 runs via chaibot earlier this morning.  should be done in a few hours.

**Comment by vr4manta** [comment]
@everettraven verify-feature-promotion is now passing.
